shell/shared/grain.js

```diff
@@ -1124,6 +1124,15 @@ if (Meteor.isClient) {
         check(path.charAt(0), '/');
         // TODO(security): More sanitization of this path. E.g. reject "/../../".
         senderGrain.setPath(path);
+      } else if (event.data.startSharing) {
+        // Allow the current grain to request that the "Share Access" menu be shown.
+        // Only show this popup if no other popup is currently active.
+        // TODO(security): defend against malicious apps spamming this call, blocking all other UI.
+        var currentGrain = getActiveGrain(globalGrains.get());
+        if (senderGrain === currentGrain && !globalTopbar._expanded.get()) {
```

Comment:
A leading underscore indicates a private member. Can you add public methods to SandstormTopbar.prototype to use here? Maybe like `isPopupOpen()` and `openPopup(name)`?